core/src/lib.rs

```diff
@@ -77,3 +80,5 @@ pub use parking_lot::{ParkResult, UnparkResult, RequeueOp, UnparkToken, ParkToke
 pub use parking_lot::{DEFAULT_UNPARK_TOKEN, DEFAULT_PARK_TOKEN};
 pub use parking_lot::{park, unpark_one, unpark_all, unpark_requeue, unpark_filter};
 pub use spinwait::SpinWait;
+#[cfg(feature = "deadlock_detection")]
+pub use parking_lot::{register_resource, deregister_resource, check_deadlock};
```

Comment:
It would also be nice if these functions we no-ops if the `deadlock_detection` feature is not defined. This avoids the need to spam `#[cfg(feature = "deadlock_detection")]` in the mutex implementations.

Reply:
Good idea.